include/vapor/SliceGridAlongPlane.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include <vapor/glutil.h>
+#include <glm/glm.hpp>
+
+class RegularGrid;
+
+struct planeDescription {
+    std::vector<double> origin;
+    std::vector<double> rotation;
+    std::vector<double> boxMin;
+    std::vector<double> boxMax;
+};
+
+//! Create a 2D grid that is sampled along the orientation of a 3D grid.
+//! The "planeDescription" variable includes parameters that define the
+//! 2D grid's origin, rotation, and extents.  The sampling rate is defined
+//! by the "sideSize" parameter, which defines how many samples along the X
+//! and Y axes are taken along the plane inside of the 3D grid.
+//!
+//! \param[in] grid3d A variable's 3D grid to be sampled along a plane, to generate a 2D grid object
+//! \param[in] description A set of std::vector<dobule> values that define the origin, rotation, and extents of the returned 2D grid
+//! \param[in] sideSize The sampling rate that the 2D grid will be defined upon, along its X and Y axes
+//! \param[out] data An array of sideSize*sideSize values that contain floating point values of the sampled data.
+//! \param[out] windingOrder A vector of double values that contain the spatial coordinates of the vertices that correspond to the two triangles that define the texture being drawn by the "data" array.
+//! \param[out] rectangle3D A set of four vertices that define the rectangle containing the returned 2D grid's texture.  Useful for debugging.
+//!
+
+VAPoR::RegularGrid* SliceGridAlongPlane(
+    const VAPoR::Grid *grid3d, 
+    planeDescription description, 
+    size_t sideSize, 
+    //std::shared_ptr<float> data, 
+    std::unique_ptr<float>& data, 
```

Comment:
Wouldn't the grid contain this data?

Reply:
No, it actually does not.  The Grid class does contain the extents of the data, but this is not have enough information to define the vertices of the texture's triangles.  Many slices can be defined within the extents of a Grid.

Reply:
There may be something I’m missing here, but would that not contradict the TwoDDataRenderer generating a textured triangle mesh from a Grid class?

Reply:
I'm not completely familiar with how the TwoD renderers work, but I do know that currently they can be only oriented on the XY plane.  I believe this restriction is what allows them to render without explicitly stating their texture vertex locations.

Reply:
Just adding a note that we're going to approve this as is (@StasJ need your approval) and if there is an opportunity to refactor to address the issue raised by Stas it can be done as part of #2728